### PR TITLE
fix sflow ipv4 and ipv6 data layer decoding

### PR DESCRIPTION
**Pull request by sharath-vemula**
according to https://sflow.org/sflow_version_5.txt, sflow sampled_ipv4 and sampled_ipv6 also has
SFlowBaseFlowRecord which is also evident from the solarwinds sflow pcap. 
Corrected the decoding logic to incorporate SFlowBaseFlowRecord

### CODE REVIEW

File: layers/sflow.go [hunk 1]
```diff
@@ -1832,6 +1832,59 @@ func decodeExtendedUserFlow(data *[]byte) (SFlowExtendedUserFlow, error) {
...
+//  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//  |      20 bit Interprise (0)     |12 bit format |
+//  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//  |                  record length                |
+//  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+//  /           Packet IP version 4 Record          /
+//  /                                               /
+//  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+
+type SFlowIpv4DataRecord struct {
```
Comment (ericyitaowu) [inline layers/sflow.go]: Can we modify the existing SFlowIpv4Record directly? Is there any concern not to touch the original code?

Comment (sharath-vemula) [reply layers/sflow.go]: Ack

File: layers/sflow_test.go [hunk 2]
```diff
@@ -13,6 +13,59 @@ import (
 	"github.com/google/gopacket"
 )
 
+// Test packet collected from PCAP from SolarWinds tool. See the test below for contents
+var SFlowTestPacketWithIPv4Data = []byte{
```
Comment (ericyitaowu) [inline layers/sflow_test.go:17]: Can you trim the test data to include IPv4 sample only, no counter sample? It should be easy to do so. You just need to modify a few bytes.

Comment (sharath-vemula) [reply layers/sflow_test.go:17]: Ack

**Comment by ericyitaowu** [comment]
looks good

**Pull request merged**